src/utils/provenance.js

```diff
@@ -40,18 +41,53 @@ function checkPermissions(npmVersion) {
 }
 
 /**
- * Fail fast and throw a meaningful error if NPM Provenance will fail silently or misleadingly.
+ * NPM does an internal check on access that fails unnecessarily for first-time publication
+ * of unscoped packages to NPM. Unscoped packages are always public, but NPM's provenance generation
+ * doesn't realise this unless it sees the status in a previous release or in explicit options.
+ */
+async function getAccessAdjustment({ access } = {}) {
+  // Don't overrule any user-set access preference.
+  if (access) return
+
+  const { name: packageName, publishConfig } = getLocalInfo()
+
+  // Don't do anything for scoped packages - those require being made public explicitly.
+  // Let NPM's own validation handle it if a user tries to get provenance on a private package.
+  // `.startsWith('@')` is what a lot of NPM internal code use to detect scoped packages,
+  // they don't export any more sophisticated scoped name detector any more.
+  if (packageName.startsWith('@')) return
+
+  // Don't do anything if the user has set any access control in package.json publishConfig.
+  // https://docs.npmjs.com/cli/v9/configuring-npm/package-json#publishconfig
+  // Let NPM deal with that internally when `npm publish` reads the local package.json file.
+  if (publishConfig?.access) return
+
+  // Don't do anything if package is already published.
+  const publishedInfo = await getPublishedInfo()
+  if (publishedInfo) return
+
+  // Set explicit public access **only** if it's unscoped (inherently public), a first publish
+  // (so we know NPM will fail to realise that this is inherently public), and the user
+  // has not attempted to explicitly set access themselves anywhere.
+  return { access: 'public' }
+}
+
+/**
+ * Fail fast and throw a meaningful error if NPM Provenance will fail silently or misleadingly,
+ * and where necessary, tweak publish options without overriding user preferences or expectations.
  *
  * @see https://docs.npmjs.com/generating-provenance-statements
- *
- * @param {string} npmVersion
  */
-function checkProvenanceViability(npmVersion) {
+async function ensureProvenanceViability(npmVersion, publishOptions) {
   if (!npmVersion) throw new Error('Current npm version not provided')
   checkIsSupported(npmVersion)
   checkPermissions(npmVersion)
-  // There are various other provenance requirements, such as specific package.json properties, but these
-  // may change in future NPM versions, and do fail with meaningful errors, so we let NPM handle those.
+
+  const value = {
+    ...publishOptions,
+    ...(await getAccessAdjustment(publishOptions)),
+  }
+  return value
```

Comment:
in order to avoid mutating the state of the called function, just use Object.assign